src/test/java/com/opencart/tests/currency/ChangeCurrencyTest.java

```diff
@@ -0,0 +1,53 @@
+package com.opencart.tests.currency;
+
+import com.opencart.enums.CurrencyName;
+import com.opencart.navigation.Navigation;
+import com.opencart.steps.AdminCurrencyPageBL;
+import com.opencart.steps.AdminDashboardBL;
+import com.opencart.steps.AdminLoginPageBL;
+import com.opencart.steps.MainPageBL;
+import com.opencart.tests.BaseTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static com.opencart.enums.URLs.ADMIN_BASE_URL;
+import static com.opencart.enums.URLs.BASE_URL;
+
+/**
+ * Test checks if the cost in EUR, USD, GBP, UAH of all products on the website's main page is correct"
+ */
+public class ChangeCurrencyTest extends BaseTest {
+    private Map<String, String> currencies;
+
+    @BeforeClass
+    public void getCurrenciesValuesFromAdminPage() {
+        new Navigation().navigateToURrl(ADMIN_BASE_URL.getValue());
+        new AdminLoginPageBL().adminLogin();
+        new AdminDashboardBL()
+                .getAdminLeftNavigationPanelPageBL()
+                .clickOnLeftNavigationPanelButton("System")
+                .clickOnLeftNavigationPanelButton("Localisation")
+                .clickOnLeftNavigationPanelButton("Currencies");
+        currencies = new AdminCurrencyPageBL().getCurrencyValues();
+    }
+
+    @DataProvider(name = "data-provider-currency")
+    public Object[][] dataProviderMethod() {
+        return new Object[][]{{CurrencyName.EUR.getValue()}, {CurrencyName.USD.getValue()},
+                {CurrencyName.GBP.getValue()}};
+    }
```

Comment:
You do not need to use this Data-Provider, you already have all necessary data into currencies map

Reply:
aaa.. I think i got it.